app/models/message.rb

```diff
@@ -58,6 +61,16 @@ def self.send_site_message(user_id, subject, message)
     msg.save
   end
 
+  def self.unread_count_for(user)
+    Rails.cache.fetch(Message.cache_string_for(user.id), expires_in: 1.month) do
```

Comment:
So, I know we should invalidate this correctly with callbacks and everything, buuuut maybe we should reduce this to like `expires_in: 1.day` so that if something does go wrong with invalidation it'll be resolved on a short time scale? We should still get a good cache hit rate, since a typical user will load the site a lot more than just once in a single day, but I'm just a bit wary of such a long cache expiry here.

Reply:
(Same for news)

Reply:
hmm, yeah, reducing that seems reasonable since it should be cleared within a day anyway.

Reply:
the same *doesn't* apply to news, though, I'm more uncertain on that.

Reply:
The main metric we probably care about for this is cache hit rate – how many times are we doing queries unnecessarily after this change? Even if we reduce the caching expiry to 1 day, we'll still only load this query once per day per user, which is a _lot_ less than our current number of page loads, so the caching still performs a great job if we drop it down to expiring daily. We could reduce it further, to once per month per user, by using this longer expiry, but as these are cheap queries, it probably doesn't help much on this margin, right?

It's not a huge deal either way, I don't think, since the invalidation is tested and looks good, though 🙂